### PR TITLE
Format: add a guide

### PR DESCRIPTION
**Pull request by squioc**
Add a guide to explain how to create format

### CODE REVIEW

File: _shared_content/develop/guides/formats/parser.md [hunk 1]
```diff
@@ -0,0 +1,658 @@
...
+    "vendor": "Security",
+    "product": "threatmanager",
+    "version": "1.0"
+  }
+}
+```
+
+## Action
+
+An action is an elementary operations that can create, update and delete fields.
```
Comment (Adamowoc) [inline _shared_content/develop/guides/formats/parser.md:543]:
```suggestion
An action is an elementary operation that can create, update and delete fields.
```

File: _shared_content/develop/guides/formats/parser.md [hunk 2]
```diff
@@ -0,0 +1,658 @@
...
+The execution of an action can be conditionned to a filter.
```
Comment (Adamowoc) [inline _shared_content/develop/guides/formats/parser.md:544]:
```suggestion
The execution of an action can be conditioned by a filter.
```

File: _shared_content/develop/guides/formats/parser.md [hunk 3]
```diff
@@ -0,0 +1,658 @@
...
+    http.response.status_code: http.response.status_message
+    api.status_code: api.status_message
+  fallback: "Request Processed"
+  filter: '{{stage1.log_type == "network"}}'
+```
+
+### References
+
+Each stage generates a layer with a set of produced fields in the stage.
+You can refer one of these field from another field, a filter or the input of a next common stage.
```
Comment (Adamowoc) [inline _shared_content/develop/guides/formats/parser.md:601]:
```suggestion
You can refer to one of these fields from another field, a filter or the input of a next common stage.
```

File: _shared_content/develop/guides/formats/parser.md [hunk 4]
```diff
@@ -0,0 +1,658 @@
...
+To refer a field, use a [jinja](https://jinja.palletsprojects.com) placeholder (ie `{{xxxx}}`) with the name of the referred layed (ie, the name of the stage) as the prefix then the path to the field in dot-notation
```
Comment (Adamowoc) [inline _shared_content/develop/guides/formats/parser.md:602]:
```suggestion
To refer to a field, use a [jinja](https://jinja.palletsprojects.com) placeholder (ie `{{xxxx}}`) with the name of the referred layer (ie, the name of the stage) as the prefix then the path to the field in dot-notation
```

File: _shared_content/develop/guides/formats/parser.md [hunk 5]
```diff
@@ -0,0 +1,658 @@
...
+(eg, to refer the value of the field `date` from the `parsed_date` stage, use `{{parsed_date.date}}`).
```
Comment (Adamowoc) [inline _shared_content/develop/guides/formats/parser.md:603]:
```suggestion
(eg, to refer to the value of the field `date` from the `parsed_date` stage, use `{{parsed_date.date}}`).
```

File: _shared_content/develop/guides/formats/smartdescriptions.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
+# How to write smart descriptions
+
+An event can be hard to read as a raw data. To make the visualization and the understand of events easier, smart-descriptions help to display important information to the user.
```
Comment (Adamowoc) [inline _shared_content/develop/guides/formats/smartdescriptions.md:3]:
```suggestion
An event can be hard to read as a raw data. To make the visualization and the understanding of events easier, smart-descriptions help displaying important information to the user.
```

File: _shared_content/develop/guides/formats/smartdescriptions.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+	}
+	```
+
+	The description `Connection from {source.ip}` will result in `Connection from 1.2.3.4`.
+
+- A list of conditions (optional):
+
+	These conditions define when the description should be displayed. For a same event, the description, with the largest number of matching conditions, will succeed.
+
+	A condition is compound of a path (dot-notation) to the field to test, and the expected value of the field (equal operation).
```
Comment (Adamowoc) [inline _shared_content/develop/guides/formats/smartdescriptions.md:38]:
```suggestion
	A condition is a compound of a path (dot-notation) to the field to test, and the expected value of the field (equal operation).
```

File: _shared_content/develop/guides/formats/smartdescriptions.md [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+	{
+	    "field": "event.kind"
+	}
+	```
+
+	will test the existence of `event.kind`
+
+- A list of relationships (optional): The relationships help to connect information from the structured event.
+
+	A relationship is compound of:
```
Comment (Adamowoc) [inline _shared_content/develop/guides/formats/smartdescriptions.md:88]:
```suggestion
	A relationship is a compound of:
```

File: _shared_content/develop/guides/formats/taxonomy.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
+# Definition of the taxomony
+
+The taxonomy of [structured events](structured_event.md) is based on the [ECS](https://www.elastic.co/guide/en/ecs/current/ecs-reference.html) specification. This taxonomy defines a set of field available for values in [parsers](parser.md).
```
Comment (Adamowoc) [inline _shared_content/develop/guides/formats/taxonomy.md:3]:
```suggestion
The taxonomy of [structured events](structured_event.md) is based on the [ECS](https://www.elastic.co/guide/en/ecs/current/ecs-reference.html) specification. This taxonomy defines a set of fields available for values in [parsers](parser.md).
```

File: _shared_content/develop/guides/formats/taxonomy.md [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+
+## Best Practices
+
+- When a raw event encompasses a datetime, parse this information with a stage date and set the result in the field `@timestamp`.
+- Define `event.category`, `event.type` and `event.kind` according to the [event categorization fields](https://www.elastic.co/guide/en/ecs/current/ecs-category-field-values-reference.html).
+- Define `event.category` and `event.type` as array.
```
Comment (Adamowoc) [inline _shared_content/develop/guides/formats/taxonomy.md:9]:
```suggestion
- Define `event.category` and `event.type` as arrays.
```

File: _shared_content/develop/guides/formats/taxonomy.md [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+- As possible, define [`event.dataset`](https://www.elastic.co/guide/en/ecs/current/ecs-event.html#field-event-dataset).
```
Comment (Adamowoc) [inline _shared_content/develop/guides/formats/taxonomy.md:10]:
```suggestion
- If possible, define [`event.dataset`](https://www.elastic.co/guide/en/ecs/current/ecs-event.html#field-event-dataset).
```

File: _shared_content/develop/guides/formats/taxonomy.md [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+- As possbile, define observer information in [`observer.*`](https://www.elastic.co/guide/en/ecs/current/ecs-observer.html).
```
Comment (Adamowoc) [inline _shared_content/develop/guides/formats/taxonomy.md:11]:
```suggestion
- If possible, define observer information in [`observer.*`](https://www.elastic.co/guide/en/ecs/current/ecs-observer.html).
```

File: _shared_content/develop/guides/formats/taxonomy.md [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+The custom taxonomy is defined in the directory `_meta/fields.yml` in the format. The document is written in [JSON](https://json.org) dialect.
+
+### Custom field
+
+!!! warning
+	Please ensure that no field from the ECS taxonomy is suitable before creating a new field.
+
+A custom field consists of three elements:
+
+- name: The name of the field. it corresponds to the path (dot-notation) of the field in the [structured event](structured_event.md).
```
Comment (Adamowoc) [inline _shared_content/develop/guides/formats/taxonomy.md:32]:
```suggestion
- name: The name of the field. It corresponds to the path (dot-notation) of the field in the [structured event](structured_event.md).
```

File: _shared_content/develop/guides/formats/overview.md [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+The `_meta` directory consists of 4 files:
+
+- A logo that represent the product
+- a `manifest.yml` file with the UUID of the format, its name, a slug, a short description and data sources (see [Trend Micro Deep Security](https://github.com/SEKOIA-IO/intake-formats/blob/main/Trend%20Micro/deep-security/_meta/manifest.yml) manifest)
+- a [taxonomy](taxonomy.md) file (e.g.: [Windows taxonomy](https://github.com/SEKOIA-IO/intake-formats/blob/main/Windows/windows/_meta/fields.yml))
+- some smart-descriptions (e.g.: [Windows smart-descriptions](https://github.com/SEKOIA-IO/intake-formats/blob/main/Windows/windows/_meta/smart-descriptions.json))
+
+## Definition of a parser
+
+The parser is a vocabulary scheduling how information will be extracted from the raw event.
```
[the remark this answers is not in the document]
Comment (squioc) [reply _shared_content/develop/guides/formats/overview.md:40]: I want to mean 'a set of word' or ' a set of instructions'.
Maybe language is more suitable.

[the remark this answers is not in the document]
Comment (Adamowoc) [reply _shared_content/develop/guides/formats/overview.md:40]: @squioc I think "a set of instructions" would be nice :+1:

**Review by ka0ula — APPROVED** [review]

**Pull request merged**